### ISSUE TITLE
s3Client settings validation

### ISSUE THREAD

**Issue by spawn-guy** [CodeSleeve/stapler#112]
Hello,
I've planned to use the "default" approach with configuring AWS
http://docs.aws.amazon.com/aws-sdk-php/guide/latest/quick-start.html#client-factory-method
they, basically, say that most of the options are "optional" or "replaceable".

but in \Codesleeve\Stapler\Validator->validateS3Options there are those checks.. that stop me from using key=null,secret=null. and, actually, they are not that much needed, if these keys are read from the environment. and if i already have a default AWS Instance.
i was thinking about using 's3_client_config' => null, to use the default config.

i think S3Client will throw an error if it was mis-configured. but stapler adds an additional level of validation.. so i must say: "These options will be NOT passed directly" but "validated and passed". 

Is it possible to remove those validators? 
or update them to the proper AWS implemented version?
any other ideas?

**Comment by tabennett** [comment]
Yeah,  think we can remove those validation checks.  When I first wrote the S3 driver I had no idea you could even store your instance profile credentials like that (via AWS IAM).  I'll see if about getting those removed.  If you want to PR it you're more than welcome to.

**Comment by sonuku** [comment]
Also noting that the key and secret have moved into a credentials array for 3.0.x of S3.
This is an annoyance now.
